### PR TITLE
feat: golang client implementation ForwardMessageToDeadLetterQueue

### PR DESCRIPTION
**Pull request by tianping526**

### CODE REVIEW

**Comment by tianping526** [comment]
fix: #468

File: golang/simple_consumer.go [hunk 1]
```diff
@@ -183,6 +184,19 @@ func (sc *defaultSimpleConsumer) wrapAckMessageRequest(messageView *MessageView)
 	}
 }
 
+func (sc *defaultSimpleConsumer) wrapForwardMessageToDeadLetterQueueRequest(messageView *MessageView) *v2.ForwardMessageToDeadLetterQueueRequest {
```
Comment (aaron-ai) [inline golang/simple_consumer.go:187]: There should not be the `ForwardMessageToDeadLetterQueueRequest` in SimpleConsumer.

File: golang/simple_consumer.go [hunk 2]
```diff
@@ -427,3 +441,27 @@ func (sc *defaultSimpleConsumer) Ack(ctx context.Context, messageView *MessageVi
 	sc.cli.doAfter(MessageHookPoints_ACK, messageCommons, duration, messageHookPointsStatus)
 	return nil
 }
+
+func (sc *defaultSimpleConsumer) ForwardMessageToDeadLetterQueue(ctx context.Context, messageView *MessageView) error {
```
Comment (aaron-ai) [inline golang/simple_consumer.go:445]: ditto

File: golang/simple_consumer.go (hunk 1 above)
Comment (tianping526) [reply golang/simple_consumer.go:187]: Is there any good way to customize the retry policy? For example, if there are two messages m and n in topic A, I want to retry 3 times for m, but 100 times for n, and the retry interval can be decided by the client

**Pull request closed without merging**